### DISCUSSION TITLE
Hot code reloading?

### DISCUSSION THREAD

**Discussion by nahuel** [DioxusLabs/dioxus#63] [General]
A big advantage of React is hot-code reloading/HMR while developing (and much faster with next gen builders like esbuild and vitejs), so you can tweak an UI incrementally without reloading the entire SPA. There are plans for this? Rust is more static than JS (and ClojureScript it's even better than JS for incremental dev), but maybe is possible to preserve/serialize the app state while reloading an entire wasm bundle and automagically adapting it to the new code version.

**Comment by jkelleyrtp** [comment]
Yes! I've been thinking we could build a small interpreter for the `rsx!` macro and allow two-way editing for source from the DOM too.

For `rsx!` code - yes definitely. Once you start to venture out into regular Rust code then it can be more challenging. With WASM I think we could patch the component's function itself, and just re-render it when its source changes, that way pointers and stuff still point properly.
